### DISCUSSION TITLE
Hierarchical transitions

### DISCUSSION THREAD

**Discussion by Perlkonig** [statelyai/xstate#1363] [General] [answered]
Here's the pertinent part of the definition:

```ts
assigning: {
    initial: "rolling",
    states: {
        rolling: {
            entry: [rollDice],
            always: ["waiting"],
        },
        waiting: {
            on: {
                ASSIGN: "processing"
            }
        },
        processing: {
            entry: [assignDie],
            always: [
                {
                    target: "..resolving",
                    cond: "searchFull"
                },
                {
                    target: "waiting",
                    cond: "stillDice",
                },
                {
                    target: "rolling",
                }
            ]
        }
    }
},
resolving: {
    entry: [(context, event) => {console.log("Made it!");}],
},
```

Everything is working until I try to transition to `..resolving`. I end up with the following error: `RangeError: Maximum call stack size exceeded`. I read the docs on "Hierarchical State Nodes," and it shows how to transition to deeper states, but it doesn't mention going shallower. The two dots make sense, and if I don't include them the interpreter throws an error, but for some reason the transition is not happening as I expect. Adding `internal: true` doesn't change anything either. What am I not getting?

Thanks for your time!

**Comment by Andarist** [comment, marked as answer]
`..` is not a valid specifier. You currently can't transition to a sibling of a parent node using any kind of relative paths.

You might be interested in the discussion happening below this PR: https://github.com/davidkpiano/xstate/pull/1246

**Reply by Perlkonig** [reply]
Odd that the interpreter doesn't blow up, and odd that it triggers some sort of infinite loop. Regardless, after RTFM *again* (I'm clearly slow), I tried [a custom ID](https://xstate.js.org/docs/guides/ids.html#custom-ids) and it worked fine. So be it. Thanks again for your time!

```ts
assigning: {
    initial: "rolling",
    states: {
        rolling: {
            entry: [rollDice],
            always: ["waiting"],
        },
        waiting: {
            on: {
                ASSIGN: "processing"
            }
        },
        processing: {
            entry: [assignDie],
            always: [
                {
                    target: "#resolving",
                    cond: "searchFull",
                    internal: true
                },
                {
                    target: "waiting",
                    cond: "stillDice",
                },
                {
                    target: "rolling",
                }
            ]
        }
    }
},
resolving: {
    id: "resolving",
    entry: [(context, event) => {console.log("Made it!");}],
},
```
